libkbfs/keybase_daemon_rpc.go

```diff
@@ -250,64 +272,54 @@ func (k *KeybaseDaemonRPC) AddProtocols(protocols []rpc.Protocol) {
 	if protocols == nil {
 		return
 	}
+
+	k.lock.RLock()
```

Comment:
this should be a `k.lock.Lock()`, right? Since you append or write to `k.protocols`.